quantmsio/core/feature.py

```
@@ -1,535 +1,260 @@
+        self._ms_runs = self.extract_ms_runs()
+        self._protein_global_qvalue_map = self.get_protein_map()
+        self._modifications = self.get_modifications()
+        self._score_names = self.get_score_names()
+        self.experiment_type = SDRFHandler(sdrf_path).get_experiment_type_from_sdrf()
+        self._mods_map = self.get_mods_map()
+        self._automaton = get_ahocorasick(self._mods_map)
+
+    def _extract_pep_columns(self):
+        if os.stat(self.mztab_path).st_size == 0:
+            raise ValueError("File is empty")
+        f = open(self.mztab_path)
```

Ruff

Use a context manager for opening files

(SIM115)